docs/writing-rules/data-flow/overview.md

```diff
@@ -5,22 +5,36 @@ description: >-
   Semgrep can run data-flow analyses on your code, this is used for constant propagation and for taint tracking.
 ---
 
-# Engine overview
+import SupportedLanguagesTable from '/src/components/reference/_supported-languages-table.mdx'
 
-Semgrep comes with an intra-procedural data-flow analysis engine that provides it with additional capabilities. [Constant propagation](../constant-propagation/) allows Semgrep to, for example, match `return 42` against `return x` when `x` can be reduced to `42` by constant folding. [Taint tracking (aka taint mode)](../taint-mode/) allows writing simple rules that catch complex [injection bugs](https://owasp.org/www-community/Injection_Flaws), such as those that can result in [cross-site scripting (XSS)](https://owasp.org/www-community/attacks/xss/).
+# Data-flow analysis engine overview
```

Comment:
Is data-flow one-to-one synonym with taint analysis, or there can be more data-flows, and taint analysis is one of them? @ievans